vision_agent_tools/models/flux1.py

```diff
@@ -0,0 +1,141 @@
+        self, prompt: str, image: Image.Image, mask_image: Image.Image
+    ) -> Image.Image:
+        """Inpaint an image from a given prompt and mask image."""
+        try:
+            generator = torch.Generator("cpu").manual_seed(0)
+            image = self._inpaint_pipeline(
+                prompt=prompt,
+                height=self.model_config.height,
+                width=self.model_config.width,
+                guidance_scale=self.model_config.guidance_scale,
+                num_inference_steps=self.model_config.num_inference_steps,
+                max_sequence_length=self.model_config.max_sequence_length,
+                generator=generator,
+                image=image,
+                mask_image=mask_image,
+                strength=self.model_config.strength,
+            ).images[0]
+            return image
+        except Exception as e:
+            print(f"Error during inpainting: {e}")
+
+    def __init__(
+        self, model_config: Flux1Config | None = None, task: str = "image-generation"
+    ):
+        """
+        Initializes the Flux1 image generation tool.
+        Loads the pre-trained Flux1 model from HuggingFace and sets the device to run the model on.
+        """
+        self.model_config = model_config or Flux1Config()
+        dtype = torch.bfloat16
+
+        self._pipeline = FluxPipeline.from_pretrained(
+            self.model_config.hf_model, torch_dtype=dtype
+        )
+        self._pipeline.enable_sequential_cpu_offload()
+
+        self._inpaint_pipeline = FluxInpaintPipeline.from_pretrained(
+            self.model_config.hf_model, torch_dtype=dtype
+        )
+        self._inpaint_pipeline.enable_sequential_cpu_offload()
+
+    @torch.inference_mode()
+    def __call__(
+        self,
+        task: Flux1Task,
+        prompt: str,
+        image: Image.Image | None = None,
+        mask_image: Image.Image | None = None,
+    ) -> Image.Image:
+        """
+        Performs object detection on an image using the Flux1 model.
+
+        Args:
+            prompts str: A prompt to generate an image from.
+                        Currently, only one prompt is supported (single string)
+                        and the model will generate a single image.
+            task Flux1Task: The task to perform using the model.
+            image Image.Image: The image to inpaint.
+            mask_image Image.Image: The mask image to use for inpainting.
+
+        Returns:
+            Flux1InferenceData:  `Flux1InferenceData` object containing the generated image.
+        """
+        if task == Flux1Task.IMAGE_GENERATION:
+            generated_img = self._generate_image(prompt=prompt)
+        elif task == Flux1Task.MASK_INPAINTING:
+            generated_img = self._inpaint_image(
+                prompt=prompt, image=image, mask_image=mask_image
+            )
+
+        return generated_img
+
+    def to(self, device: Device):
+        self._model.to(device=device.value)
```

Comment:
Looks like `self._model` is undefined here, the probably `_pipeline`?

Reply:
Resolved!